### PR TITLE
Add support for POST and GET of /integrations

### PR DESCRIPTION
**Pull request by ashwin1111**

### CODE REVIEW

File: apps/integrations/models.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    type = models.CharField(max_length=255, help_text='Type of Integration', choices=INTEGRATION_TYPE)
+    is_active = models.BooleanField(default=False, help_text='Is Integration Active')
+    is_beta = models.BooleanField(default=False, help_text='Is Beta')
+    connected_at = models.DateTimeField(auto_now_add=True, help_text='Connected at datetime')
+    disconnected_at = models.DateTimeField(null=True, help_text='Disconnected at datetime')
+    updated_at = models.DateTimeField(auto_now=True, help_text='Updated at datetime')
+
+    class Meta:
+        db_table = 'integrations'
+        unique_together = ('org_id', 'type')
```
Comment (Shwetabhk) [inline apps/integrations/models.py:20]: This means only 1 type will be there in the DB for 1 org and for demo orgs we'll probably overwrite this correct?
If yes then cool

File: apps/integrations/views.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+from .serializers import IntegrationSerializer
+
+
+logger = logging.getLogger(__name__)
+logger.level = logging.INFO
+
+
+
+class IntegrationsView(generics.ListCreateAPIView):
+    authentication_classes = []
```
Comment (Shwetabhk) [inline apps/integrations/views.py:18]: can we remove this?

File: apps/integrations/views.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    permission_classes = []
```
Comment (Shwetabhk) [inline apps/integrations/views.py:19]: Permission class will be null only right?

File: apps/integrations/views.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    serializer_class = IntegrationSerializer
+    queryset = Integration.objects.filter(is_active=True, is_beta=True)
+    filterset_fields = {'type': {'exact'}}
+
+    def perform_create(self, serializer):
```
Comment (Shwetabhk) [inline apps/integrations/views.py:24]: how is this function triggered?

**Review by Shwetabhk — APPROVED** [review]

File: apps/integrations/models.py (hunk 1 above)
Comment (ashwin1111) [reply apps/integrations/models.py:20]: we'll preserve the first connected org, will skip additional incoming POST for same org (incase of test orgs)

File: apps/integrations/views.py (hunk 4 above)
Comment (ashwin1111) [reply apps/integrations/views.py:24]: Screenshot 2023-08-30 at 11 50 09 AM
Screenshot 2023-08-30 at 11 50 23 AM


We are overriding this

File: apps/integrations/views.py (hunk 2 above)
Comment (ashwin1111) [reply apps/integrations/views.py:18]: will remove this

File: apps/integrations/views.py (hunk 4 above)
Comment (ashwin1111) [reply apps/integrations/views.py:24]: We would need request header to be able to extract access token and call few apis and get org_id. We cannot do that with signals and serializer. Found this way to do it neatly

File: apps/integrations/models.py (hunk 1 above)
Comment (Shwetabhk) [reply apps/integrations/models.py:20]: cool

File: apps/integrations/views.py (hunk 4 above)
Comment (Shwetabhk) [reply apps/integrations/views.py:23]: Nice

**Review by NileshPant1999 — APPROVED** [review]

**Pull request merged**